indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -19,48 +24,125 @@ import kotlinx.serialization.encoding.Encoder
  * @throws Asn1Exception if less than two nodes are supplied, the first node is >2 or the second node is >39
  */
 @Serializable(with = ObjectIdSerializer::class)
-class ObjectIdentifier @Throws(Asn1Exception::class) constructor(@Transient vararg val nodes: UInt) :
+class ObjectIdentifier @Throws(Asn1Exception::class) private constructor(
+    bytes: ByteArray?,
+    nodes: List<BigInteger>?
+) :
     Asn1Encodable<Asn1Primitive> {
-
     init {
-        if (nodes.size < 2) throw Asn1StructuralException("at least two nodes required!")
-        if (nodes[0] * 40u > UByte.MAX_VALUE.toUInt()) throw Asn1Exception("first node too lage!")
-        //TODO more sanity checks
+        if ((bytes == null) && (nodes == null)) {
+            //we're not even declaring this, since this is an implementation error on our end
+            throw IllegalArgumentException("either nodes or bytes required")
+        }
+        if (bytes?.isEmpty() == true || nodes?.isEmpty() == true)
+            throw Asn1Exception("Empty OIDs are not supported")
+
+        bytes?.apply {
```

Comment:
maybe move this to the `lazyOf` initializer?

`lazyOf(bytes.also { /* .. validate .. */ })`

Reply:
nope; I want it to break right at the initialization, not some time later on

Reply:
that _would_ break right at the initialization

Reply:
seems so. Still no, because i want all checks in one place to see them at a glance